src/Purl/Fragment.php

```diff
@@ -109,7 +109,16 @@ public function __toString() : string
     protected function doInitialize() : void
     {
         if ($this->fragment !== null) {
-            $this->data = array_merge($this->data, parse_url($this->fragment));
+            $pos = strpos($this->fragment, ':', 1);
```

Comment:
Hi @jwage, I use the ```strpos``` function to get the first position of the colon character.
And sub the string with colon then return the current url with the valid fragment.

Reply:
I think we misunderstood each other. It should maintain the whole hashbang. i.e. `#hello:123`

Reply:
You're right. The hash bang value is the ```#hello:123```.
Perhaps we can rebuild the fragment value after using the ```parse_url``` function.

What do you think?

Reply:
Yes. We would need to parse out the hashbang manually instead of relying on parse_url since it can't handle it when it has a colon on it.